### PR TITLE
✨ Add logger pipe command

### PR DESCRIPTION
**Pull request by schitcrafter**
Implements #159

### CODE REVIEW

File: src/bot/checks.rs
Comment (maxwai) [inline src/bot/checks.rs:62]: I don't think None is very useful here as this should just be interpreted as false. The Methods calling this shouldn't actually care why a check returns false, only that the user doesn't have the permission to do so.

File: src/bot/commands.rs
Comment (maxwai) [inline src/bot/commands.rs:85]: if not log pipe was setup, the user will not get any feedback from the command

**Review by maxwai — CHANGES_REQUESTED** [review]
A few small things.

Also after looking at the code it doesn't really make sense to allow the main log pipe to be changed, it should just always be the first channel. Not a lot needs to change, only the methods to change the log pipe and the not needed `sent_in_main_guild` calls. The `mysql_lib::get_guild` call doesn't need additional checks since the main guild is never in the database.

**Review by maxwai — COMMENTED** [review]
looks good, will not merge yet since I want to test it first but can't right now

**Comment by maxwai** [comment]
After trying the code, the log messages are not send in Discord. I found that it doesn't work when used inside `block_on` (don't know why), but wrapping it in a `spawn_blocking` makes it work.
@schitcrafter look at the changes and merge if the changes are fine for you

**Review by maxwai — APPROVED** [review]

**Comment by schitcrafter** [comment]
Gonna take a guess that tracing didn't like blocking in it's layer (it's calling from an async context, so tokio, which also doesn't like blocking), that's probably why spawning is necessary. Probably should've been that from the beginning to be honest.

For the `LOG_LEVEL` thing, I was under the impression that fmt already comes with an EnvFilter (it only does when using its `init()` function) - but I still think that we should use [EnvFilter](https://docs.rs/tracing-subscriber/latest/tracing_subscriber/filter/struct.EnvFilter.html). You can set the log level like you do manually here, but you also have much finer control about what to log (like switching to trace only for some module/crate/function/whatever). That's very nice for debugging, in my opinion.

**Comment by schitcrafter** [comment]
@maxwai Changes look good but I'd at least like to know if we want the EnvFilter or not, then we can merge this (plus #199 and after some testing #201 )

**Comment by maxwai** [comment]
The Env Filter is not necessary here. we can still switch to trace using the env for the console and file outputs

**Comment by schitcrafter** [comment]
Yeah okay but...that's not why I'd use EnvFilter. I'd use it because if, for example, I wanted to debug something with subject commands, and switched to trace globally for that, I'd have tons of unnecessary output. With EnvFilter I could cut down on that by only specifying trace for that specific module. But if you don't wanna merge it

**Pull request merged**
